zebra-rpc/src/server/tests/vectors.rs

```diff
@@ -23,6 +23,7 @@ use super::super::*;
 
 /// Test that the JSON-RPC server spawns when configured with a single thread.
 #[test]
+#[cfg(not(target_os = "windows"))]
 fn rpc_server_spawn_single_thread() {
```

Comment:
Optional/Nitpick: This test worked for me locally if we want to instead disable the `rpc_sever_spawn_parallel_threads()` on Windows and run this one? 

I think it's fine either way, but:
- If we leave the parallel one enabled, we'll be able to check if a warning is removed in a future version of Zebra
- If we enable this one instead, the test name will be more accurate if it's falling back on a single-threaded server.

I don't think it's important to run both tests on Windows, and I suspect there'll be slightly more frequent port conflicts in the tests on Windows if both are enabled because it uses a different [`random_known_port()`](https://github.com/ZcashFoundation/zebra/blob/test_zcash_script_v0.1.15/zebra-test/src/net.rs#L63).

Optional/Nitpick: Could we also fix the typo in the name? (it should be "rpc_server_spawn_parallel_threads", it's been there for a while, doesn't feel like it's worth its own PR but it would make it easier to search through the code or test results).

Not important, but, interestingly, it's logging that, ["Multi-threaded server is not available on Windows. Falling back to single thread."](https://github.com/ZcashFoundation/zebra/actions/runs/8653751415/job/23729618725#step:12:4667). I haven't found the log in Zebra's code or the relevant dependencies, I think I saw a merged PR in the JSON-RPC github repo for supporting multi-threaded servers on Windows, I'm not sure what's happening there.

I don't think we need to support multi-threaded RPC servers on Windows in Zebra for now, we could open an issue for re-enabling the test if we move away from JSON-RPC (and potentially reading the port from return value of `RpcServer::spawn` to avoid port conflicts).

Reply:
Disabling multi threads and leaving single ones seems to fail: https://github.com/ZcashFoundation/zebra/actions/runs/8784436404/job/24102673174?pr=8393

I reverted back to disabling single thread test in https://github.com/ZcashFoundation/zebra/pull/8393/commits/941a082de1a63fe2c82c69739baa5a5de9cac710